first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -102,9 +122,120 @@ public ContainerBuildPlan extendContainerBuildPlan(
         .map(FileEntriesLayer.Builder::build)
         .filter(layer -> !layer.getEntries().isEmpty())
         .forEach(newPlanBuilder::addLayer);
+
+    ContainerBuildPlan newPlan = newPlanBuilder.build();
+
+    if (config.get().isCreateParentLayers()) {
+      newPlan = moveParentDepsToNewLayers(newPlan, mavenData, logger);
+    }
+
+    return newPlan;
+  }
+
+  private ContainerBuildPlan moveParentDepsToNewLayers(
+      ContainerBuildPlan buildPlan, MavenData mavenData, ExtensionLogger logger)
+      throws JibPluginExtensionException {
+
+    logger.log(LogLevel.LIFECYCLE, "Moving parent dependencies to new layers.");
+
+    // the key is the expected path for the parent dependency
+    Map<String, Artifact> parentDependencies =
+        getParentDependencies(mavenData)
+            .stream()
+            .map(d -> d.getArtifact())
+            // TODO: configurable?
+            .collect(
+                // TODO: getVersion or getBaseVersion? check in jib, which version it is...
+                Collectors.toMap(
+                    a -> "/app/libs/" + a.getArtifactId() + "-" + a.getVersion() + ".jar", a -> a));
+
+    // parent dependencies that have not been found in any layer (due to different version or
+    // filtering)
+    Map<String, Artifact> parentDependenciesNotFound = new HashMap<>(parentDependencies);
+
+    List<FileEntriesLayer.Builder> newLayers = new ArrayList<>();
+
+    @SuppressWarnings("unchecked")
+    List<FileEntriesLayer> originalLayers = ((List<FileEntriesLayer>) buildPlan.getLayers());
+    originalLayers.forEach(
+        l -> {
+          // for each layer, create a parent layer
+          String parentLayerName = l.getName() + "-parent";
+          FileEntriesLayer.Builder toParentLayerBuilder =
```

Comment:
"to" layers made sense in the original code, as [the configuration name was "to"](https://github.com/GoogleContainerTools/jib-extensions/blob/1a79b4d5121be5f09679f1f387f8b2edecf556b2/first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/Configuration.java#L52-L55). However, I think it's confusing to call "to" here. How about just `parentLayerBuilder`?

Reply:
Ok. Also renamed toLayerBuilder to layerBuilder